drivers/lm75/lm75.c

```diff
@@ -0,0 +1,399 @@
+    }
+    int16_t temp_hyst_short, temp_os_short;
+    /* getting into the correct precision value in units of 10 */
+    temp_hyst = (temp_hyst * dev->lm75_params.res->os_mult) / 1000;
+    temp_os   = (temp_os * dev->lm75_params.res->os_mult) / 1000;
+
+    /* temp must first be converted to 16 bit format, and sampled to its resolution
+     * then shifted by the number of unused bits and finally reversed
+     * into little endian for writing into the register.
+     * NOTE: values smaller than the resolution steps are truncated */
+    temp_hyst_short = (int16_t) (temp_hyst / dev->lm75_params.res->os_res);
+    temp_hyst_short = temp_hyst_short << dev->lm75_params.res->os_shift;
+    temp_hyst_short = ntohs(temp_hyst_short);
+    temp_os_short   = (int16_t) (temp_os / dev->lm75_params.res->os_res);
+    temp_os_short   = temp_os_short << dev->lm75_params.res->os_shift;
+    temp_os_short   = ntohs(temp_os_short);
+
+    if (i2c_write_regs(I2C_BUS, I2C_ADDR, LM75_THYST_REG, &temp_hyst_short, 2, 0) != 0) {
+        i2c_release(I2C_BUS);
+        LOG_ERROR("ERROR wrtiting Hyst temp\n");
+        return LM75_ERROR_I2C;
+    }
+
+    if (i2c_write_regs(I2C_BUS, I2C_ADDR, LM75_TOS_REG, &temp_os_short, 2, 0) != 0) {
+        i2c_release(I2C_BUS);
+        LOG_ERROR("ERROR writing OS temp\n");
+        return LM75_ERROR_I2C;
+    }
+
+    i2c_release(I2C_BUS);
+    return LM75_SUCCESS;
+}
+
+int lm75_get_os_temp(lm75_t *dev, int *temperature) {
+
+    int16_t temp;
+    if (i2c_acquire(I2C_BUS) != 0) {
+        return LM75_ERROR_I2C;
+    }
+    /* read the temperature register */
+    if (i2c_read_regs(I2C_BUS, I2C_ADDR, LM75_TOS_REG, &temp, 2, 0) != 0) {
+        i2c_release(I2C_BUS);
+        return LM75_ERROR_I2C;
+    }
+
+    i2c_release(I2C_BUS);
+
+    /* since the value is read in little endian, it must be converted to big endian
+    * then it must be multiplied by its resolution
+    * and then shifted by the number of unused bits that must be discarded */
+    *temperature = (int16_t)htons(temp) * dev->lm75_params.res->os_res \
+                   >> dev->lm75_params.res->os_shift;
+
+    return LM75_SUCCESS;
+}
+
+int lm75_get_hyst_temp(lm75_t *dev, int *temperature) {
+
+    int16_t temp;
+    if (i2c_acquire(I2C_BUS) != 0) {
+        return LM75_ERROR_I2C;
+    }
+
+    /* read the temperature register */
+    if (i2c_read_regs(I2C_BUS, I2C_ADDR, LM75_THYST_REG, &temp, 2, 0) != 0) {
+        i2c_release(I2C_BUS);
+        return LM75_ERROR_I2C;
+    }
+
+    i2c_release(I2C_BUS);
+    *temperature = (int16_t)htons(temp) * dev->lm75_params.res->os_res \
+                   >> dev->lm75_params.res->os_shift;
+
+    return LM75_SUCCESS;
+}
+
+int lm75_get_os_pin(lm75_t *dev) {
+
+    if (i2c_acquire(I2C_BUS) != 0) {
+        return LM75_ERROR_I2C;
+    }
+
+    uint8_t config, os_pin = 0;
+
+    if (i2c_read_reg(I2C_BUS, I2C_ADDR, LM75_CONF_REG, &config, 0) != 0) {
+        i2c_release(I2C_BUS);
+        return LM75_ERROR;
+    }
```

Comment:
Couldn't we get that from `dev->lm75_params.polarity`?

Reply:
We can, but should we?

Reply:
It saves us an I2C transaction